apps/drec-api/src/pods/device/device.service.ts

```diff
@@ -70,4 +72,55 @@ export class DeviceService {
     currentDevice.status = DeviceStatus.Submitted;
     return await this.repository.save(currentDevice);
   }
+
+  private getFilteredQuery(filterDto: FilterDTO): FindManyOptions<Device> {
+    const {
+      fuel_code,
+      device_type_code,
+      installation_configuration,
+      capacity,
+      startDate,
+      endDate,
+      grid_interconnection,
+      off_taker,
+      sector,
+      labels,
+      standard_compliance,
+    } = filterDto;
+    let where: FindConditions<Device> = {};
+    if (fuel_code) {
+      where.fuel_code = fuel_code;
+    }
+    if (device_type_code) {
+      where.device_type_code = device_type_code;
+    }
+    if (installation_configuration) {
+      where.installation_configuration = installation_configuration;
+    }
+    if (capacity) {
+      where.capacity = capacity;
+    }
+    if (grid_interconnection) {
+      where.grid_interconnection = grid_interconnection;
+    }
+    if (off_taker) {
+      where.off_taker = off_taker;
+    }
+    if (sector) {
+      where.sector = sector;
+    }
+    if (labels) {
+      where.labels = ILike(`%${labels}%`);
+    }
+    if (standard_compliance) {
+      where.standard_compliance = standard_compliance;
+    }
+    if (startDate && endDate) {
+      where.commissioning_date = Between(startDate, endDate);
+    }
+    const query: FindManyOptions<Device> = {
```

Comment:
if you don't like syntax in this function I guess you may be interested in functions cleaning objects. I wrote one for building queries in mongodb and it improved readibility a lot. This works similar: https://www.npmjs.com/package/clean-deep

So you would have

```ts
const where  = cleanDeep({
   fuel_code: filter.fuel_code,
   device_type_code: filter.device_type_code,
   cinussioning_date: filter.startDate && filter.endDate && Between(startDate, endDate),
...
});
```

up to you. Advantage of this approach is that you clearly see the query you are sending

Reply:
Nice idea. I implemented it